pylint/checkers/variables.py

```diff
@@ -2494,6 +2478,31 @@ def _has_homonym_in_upper_function_scope(
             for _consumer in self._to_consume[index - 1 :: -1]
         )
 
+    @staticmethod
+    def _has_homonym_in_comprehension_test(node: nodes.Name) -> bool:
+        """
+        Return True if `node`'s frame contains a comprehension having
+
+        an "if" test employing an identical name. The use of the name
+        in the if test could come in several levels of nesting:
+
+        Examples:
+            [x for x in range(3) if name]
+            [x for x in range(3) if name.num == 1]
+            [x for x in range(3)] if call(name.num)]
+        """
```

Comment:
`pydocstringformatter` won't let me remove this blank line. i'm github-screengrabbing-it with this comment.